### PR TITLE
Always set lightweight children

### LINKED ISSUE #2250: Jenkins: Test children should have the same SCM config settings as parent
**Describe the bug**
Recently, the option for lightweight checkout was added (#2216 ). This setting, and all scm settings, should propagate down to any child jobs that are generated. This will further reduce disk space burden on the Jenkins node. It will also eliminate any surprise differences between the parent and child jobs.

### PR DESCRIPTION
**Pull request by AdamBrousseau**
- Except for grinder jobs
- Also allow Repo and Branch to be passed to DSL
- Whitespace cleanup including replacing some spaces
  with tabs to match the rest of the file

Fixes #2250

Signed-off-by: Adam Brousseau <adam.brousseau88@gmail.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
@llxia for review. Hide whitespace changes for the real changes. Untested at this point.

**Comment by AdamBrousseau** [comment]
I added an exception for Grinder jobs as this would be the likely place where we would hit an edge case.

**Comment by llxia** [comment]
Thanks @AdamBrousseau yes, Grinder is for debugging purposes, and users are very likely to change Adopt repo and branch when using Grinder. If the Grinder child jobs are generated with hardcoded Adopt repo and branch, there is a chance to rerun a Grinder child job based on the other poeple's repo and branch. Once this happens, it will very hard to detect. Due to this reason, it is better to make an exception case for Grinder child jobs.

**Comment by llxia** [comment]
On the second thought, this change should not be needed. We just need to add `LIGHT_WEIGHT_CHECKOUT` as a job parameter in https://github.com/AdoptOpenJDK/openjdk-tests/blob/master/buildenv/jenkins/testJobTemplate#L188 and any child job will inherit this parameter value from its parent.

Please see the example in the following jobs:
Grinder/13840/
Grinder_testList_1/configure

**Comment by AdamBrousseau** [comment]
I suppose that could work too. It does add yet another parameter to the test job. The only caveat I can think of is if someone has their own grinder job with hardcoded custom repo/branch and lightweight=true. That would generate a child job with lightweight=true and hardcoded to Adopt repos.

**Comment by llxia** [comment]
`LIGHT_WEIGHT_CHECKOUT ` should always be false for Grinder. We can add a note for this or remove this parameter from the Grinder. By default `LIGHT_WEIGHT_CHECKOUT` is set to false in testJobTemplate.

Personally, I would prefer less code in Jenkins script if possible.

**Comment by AdamBrousseau** [comment]
Ok. I've updated the change, please review.

**Review by llxia — COMMENTED** [review]
Thanks @AdamBrousseau LGTM. Could you also remove buildenv/jenkins/JenkinsfileBase? We can have another PR for fixing the indentation.

File: buildenv/jenkins/JenkinsfileBase [hunk 1]
```diff
@@ -333,6 +333,8 @@ def createJob( TEST_JOB_NAME, ARCH_OS ) {
 
 	def templatePath = 'openjdk-tests/buildenv/jenkins/testJobTemplate'
 
+	jobParams.put('LIGHT_WEIGHT_CHECKOUT', LIGHT_WEIGHT_CHECKOUT)
```
Comment (AdamBrousseau) [inline buildenv/jenkins/JenkinsfileBase:336]: @llxia this change is still here. Do you want the whitespace changes in a separate pr?

Comment (llxia) [reply buildenv/jenkins/JenkinsfileBase:336]: ah, sorry, I missed this line. Yes, please keep the whitespace changes.

**Review by llxia — APPROVED** [review]
Thanks @AdamBrousseau for testing this internally.

**Review by smlambert — APPROVED** [review]
thanks @AdamBrousseau !

**Pull request merged**